cpp/src/arrow/flight/sql/odbc/flight_sql/include/flight_sql/ui/add_property_window.h

```diff
@@ -0,0 +1,125 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <odbcabstraction/platform.h>
+#include "ui/custom_window.h"
+
+namespace driver {
+namespace flight_sql {
+namespace config {
+/**
+    * Add property window class.
+    */
+class AddPropertyWindow : public CustomWindow
+{
+    /**
+        * Children windows ids.
+        */
+    struct ChildId
+    {
+        enum Type
+        {
+            KEY_EDIT = 100,
+            KEY_LABEL,
+            VALUE_EDIT,
+            VALUE_LABEL,
+            OK_BUTTON,
+            CANCEL_BUTTON
+        };
+    };
+
+public:
+    /**
+        * Constructor.
+        *
+        * @param parent Parent window handle.
+        */
+    explicit AddPropertyWindow(Window* parent);
+
+    /**
+        * Destructor.
+        */
+    virtual ~AddPropertyWindow();
+
+    /**
+        * Create window in the center of the parent window.
+        */
+    void Create();
+
+    /**
+    * @copedoc ignite::odbc::system::ui::CustomWindow::OnCreate
```

Comment:
These @copedoc tags are causing doxygen failures in CI:
https://github.com/apache/arrow/actions/runs/8518121914/job/23329735731?pr=40939#step:6:3040